File: java/behavioral_pattern/strategy/PayPalPayment.java
package strategy;

public class PayPalPayment implements PaymentStrategy {
    private String email;
    private String password;

    public PayPalPayment(String email, String password) {
        this.email = email;
        this.password = password;
    }

    @Override
    public void pay(int amount) {
        System.out.println("Paid $" + amount + " using PayPal");
        System.out.println("PayPal Email: " + email);
        System.out.println("Authentication successful");
    }
}

File: typescript/behavioral_pattern/src/strategy/PayPalPayment.ts
import PaymentStrategy from "./PaymentStrategy";

class PayPalPayment implements PaymentStrategy {
    private email: string;
    private password: string;

    public constructor(email: string, password: string) {
        this.email = email;
        this.password = password;
    }

    public pay(amount: number): void {
        console.log(`Paid ${amount} using PayPal.`);
        console.log(`PayPal Email: ${this.email}`);
        console.log(`Authentication successful`);
    }
}

export default PayPalPayment;